### PR TITLE
debugging fundamental fixed test cases

### PR DESCRIPTION
**Pull request by Tasleem-web**

### CODE REVIEW

File: Debugging Fundamentals/Task1/Product.cs [hunk 1]
```diff
@@ -7,6 +9,19 @@ public Product(string name, double price)
             Name = name;
             Price = price;
         }
+        public override bool Equals(object obj)
+        {
+            if (obj == null || obj.GetType() != this.GetType())
+                return false;
+            Product other = (Product)obj;
```
Comment (maheshgali) [inline Debugging Fundamentals/Task1/Product.cs]: Please provide meaningful names to all variables.

File: Debugging Fundamentals/Task1/Utilities.cs [hunk 2]
```diff
@@ -10,16 +10,19 @@ public static class Utilities
         /// <param name="numbers">Numbers to sort.</param>
         public static void Sort(int[] numbers)
         {
+            if (numbers == null)
```
Comment (maheshgali) [inline Debugging Fundamentals/Task1/Utilities.cs:14]: check for length of numbers as well to avoid errors.

**Review by maheshgali — CHANGES_REQUESTED** [review]

File: Debugging Fundamentals/Task1/Product.cs (hunk 1 above)
Comment (Tasleem-web) [reply Debugging Fundamentals/Task1/Product.cs]: resolved

File: Debugging Fundamentals/Task1/Product.cs [hunk 3]
```diff
@@ -7,6 +9,19 @@ public Product(string name, double price)
             Name = name;
             Price = price;
         }
+        public override bool Equals(object productObject)
```
Comment (maheshgali) [inline Debugging Fundamentals/Task1/Product.cs]: incomingProduct looks good.
Don't use words like 'Object', 'class' in names. Names should look meaningful and real.

**Review by maheshgali — CHANGES_REQUESTED** [review]

File: Debugging Fundamentals/Task1/Product.cs (hunk 3 above)
Comment (Tasleem-web) [reply Debugging Fundamentals/Task1/Product.cs]: fixed
image

**Review by maheshgali — APPROVED** [review]

**Pull request merged**